bot/utils/messages.py

```diff
@@ -31,3 +34,44 @@ def santa_message(message: str, sender: nextcord.User, show_name=False):
     icon_url = sender.avatar.url if sender.avatar and show_name else None
     embed.set_author(name=name, icon_url=icon_url)
     return embed
+
+def get_random_angela_gif():
+    angela_gifs = [
+        "https://media.giphy.com/media/MF41YrnoSgZEY/giphy.gif",
+        "https://media.giphy.com/media/3ohs4kPMizP0sn727e/giphy.gif",
+        "https://media.giphy.com/media/26gspGYxmtaUUTq7u/giphy.gif",
+        "https://media.giphy.com/media/l0ExghDSRxU2g55sc/giphy.gif",
+        "https://media.giphy.com/media/l0ExiVMKsnIy0tzBC/giphy.gif",
+        "https://media.giphy.com/media/36fSgmwvt8aEpD9w7w/giphy.gif",
+    ]
+    return random.choice(angela_gifs)
+
+def birthday_message():
+    embed = nextcord.Embed(color=SUCCESS_COLOR, title='Happiest birthday from Angela Lansbury 9000!')
+    embed.set_image(url=get_random_angela_gif())
+    return embed
+
+def get_birthday_number(year: int):
+    current_year = nextcord.utils.utcnow().year
+    birthday_number = current_year - year
+    if birthday_number <= 0:
+        return "0th"
+    SUFFIXES = {1: 'st', 2: 'nd', 3: 'rd'}
+    if 10 <= birthday_number % 100 <= 20:
+        suffix = 'th'
+    else:
+        suffix = SUFFIXES.get(birthday_number % 10, 'th')
+    return str(birthday_number) + suffix
+
+def birthday_entry(embed: nextcord.Embed, birthday: Birthday, member: nextcord.Member):
+    embed.add_field(name='\u200b', value=f'{member.mention} a very happy {get_birthday_number(birthday.year)} birthday to {birthday.name.title()}!', inline=False)
+    return embed
+
+def get_special_birthday_fields(embed: nextcord.Embed):
+    # if today is October 16th
+    if nextcord.utils.utcnow().strftime('%m-%d') == '10-16':
+        embed.add_field(name='\u200b', value=f'And a very happy {get_birthday_number(1925)} birthday to dearest Dame Angela Brigid Lansbury, may she rest in peace!', inline=False)
+    # if today is March 1st and it's not a leap year
+    if nextcord.utils.utcnow().strftime('%m-%d') == '03-01' and not nextcord.utils.utcnow().year % 4 == 0:
```

Comment:
We could probably actually query for feb 29 birthdays when doing the birthday query if it's march 1st and not a leap year, I was just getting lazy about it. Next year is a leap year so we have a little over a year to change it before then 😂

Reply:
Classic problem for future us